driver/pace/driver/state.py

```diff
@@ -175,38 +194,160 @@ def _overwrite_state_from_restart(
     return state
 
 
+def _overwrite_state_from_fortran_restart(
+    path: str,
+    communicator: pace.util.CubedSphereCommunicator,
+    state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    is_gpu_backend: bool,
+):
+    """
+    Args:
+        path: path to restart files
+        rank: current rank number
+        state: an empty state
+        restart_file_prefix: file prefix name to read
+
+    Returns:
+        state: new state filled with restart files
+    """
+    state_dict = _driver_state_to_dict(state)
+
+    state_dict = pace.util.open_restart(path, communicator)
```

Comment:
state_dict is overwritten without ever being used.
```suggestion
    state_dict = pace.util.open_restart(path, communicator)
```